capstone/src/test/java/com/google/sps/objects/ChallengeTest.java

```diff
@@ -0,0 +1,123 @@
+package com.google.sps.objects;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.Objects.Badge;
+import com.google.sps.Objects.Challenge;
+
+/**
+ * Unit tests for {@link Challenge}.
+ *
+ * @author lucyqu
+ */
+public class ChallengeTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig()
+              .setDefaultHighRepJobPolicyUnappliedJobPercentage(0));
+
+  private Challenge challenge;
+  private Badge badge;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    badge = new Badge(/* challenge name */ "Run 3 miles", /* icon */ null, /* timestamp */ 234567);
+    challenge =
+        new Challenge(
+            "Run 3 miles", /* challenge name */
+            123456, /* due date */
+            badge, /* badge */
+            new ArrayList<String>(), /* completed users */
+            100 /* challenge id */);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    challenge = null;
+  }
+
+  @Test
+  public void getDueDateTest() {
+    assert challenge.getDueDate() == 123456;
+  }
+
+  @Test
+  public void getChallengeName() {
+    assertEquals(challenge.getChallengeName(), "Run 3 miles");
```

Comment:
nit: I think it might be good to declare these (and any other challenge fields) as constants up at the top, so the assert would look like `challenge.getDueDate() == DUE_DATE`. just personal preference but I think it makes the tests a bit clearer. and if for some reason you want a different challenge name, due date, id, etc, you only have to change everything in one place